### PR TITLE
samples: net: http_server: Add netusb support

### PR DESCRIPTION
**Pull request by jukkar**
Add overlay file for netusb configuration, and usb network initialization to the code.

### CODE REVIEW

**Review by rlubos — APPROVED** [review]

**Review by kartben — APPROVED** [review]
Great - thanks! Would be nice to document the available overlays (802.15.4, usb, ...) in the README at some point?

File: samples/net/sockets/http_server/src/main.c [hunk 1]
```diff
@@ -375,8 +375,19 @@ static void setup_tls(void)
...
+static inline int init_usb(void)
+{
+	return 0;
+}
+#endif /* CONFIG_USB_DEVICE_STACK */
+
 int main(void)
 {
+	init_usb();
+
```
Comment (pdgendt) [inline samples/net/sockets/http_server/src/main.c:390]: Shorter?

```c
int main(void)
{
#if defined(CONFIG_USB_DEVICE_STACK)
	init_usb();
#endif /* CONFIG_USB_DEVICE_STACK */

```

**Review by pdgendt — APPROVED** [review]

File: samples/net/sockets/http_server/src/main.c (hunk 1 above)
Comment (jukkar) [reply samples/net/sockets/http_server/src/main.c:390]: Why not, the code was copied verbatim from echo-server but we can do it like this too.

Comment (jukkar) [reply samples/net/sockets/http_server/src/main.c:390]: Hmm, actually this is not much shorter as the function needs to be introduced too. I think I will leave this one as is for now.

**Comment by jukkar** [comment]
> Would be nice to document the available overlays (802.15.4, usb, ...) in the README at some point?

Sure, I will prepare a separate PR for that.

**Pull request merged**